compiler/src/dotty/tools/dotc/typer/Scala2Macros.scala

```diff
@@ -0,0 +1,27 @@
+package dotty.tools.dotc.typer
+
+import dotty.tools.dotc.ast._
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.core._
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Decorators._
+import dotty.tools.dotc.core.Symbols._
+
+trait Scala2MacroDeclatration { self: Typer =>
```

Comment:
Not sure it's worth to split this out to a separate trait. Why not have it in `Typer`?

Reply:
Because the the logic to check if the macro has the correct signature will be relatively large. I assume that when we implement everithing, it will be too large to be in typed.

Reply:
I am  not sure it's worth to do it at all

 - the scheme as is works, it's just that errors are detected later
 - it's transitional, only needed for 2x3 cross building

So, why sink more time than necessary in

Reply:
I'll move it to typer